pygeoapi/linked_data.py

```diff
@@ -171,35 +171,89 @@ def geojson2geojsonld(config, data, dataset, identifier=None):
     :param identifier: item identifier (optional)
+    :param identifier_field: item identifier_field (optional)
 
     :returns: string of rendered JSON (GeoJSON-LD)
     """
     context = config['resources'][dataset].get('context', [])
-    data['id'] = (
-        '{}/collections/{}/items/{}' if identifier
-        else '{}/collections/{}/items'
-    ).format(
-        *[config['server']['url'], dataset, identifier]
-    )
+    geojsonld = config['resources'][dataset].get('geojsonld', True)
+
+    if identifier:
+    # Single geojsonld
+        if not geojsonld:
+            data, geocontext = make_jsonld(data)
+            data[identifier_field] = identifier
+        else:
+            data['id'] = identifier
+    else:
+    # Collection of geojsonld
+        data['@id'] = '{}/collections/{}/items/'.format(
+            config['server']['url'], dataset)
+        for i, feature in enumerate(data['features']):
+            identifier = feature.get(identifier_field, feature.get('properties').get(identifier_field, ''))
+            if not is_url(str(identifier)):
+                identifier = '{}/collections/{}/items/{}'.format(
+                            config['server']['url'], dataset, feature['id'])
+
+            if not geojsonld:
+                feature, geocontext = make_jsonld(feature)
+                # Note: @id or https://schema.org/url or both or something else?
+                feature[identifier_field] = identifier
+            else:
+                feature['id'] = identifier
+           
+            data['features'][i] = feature
+
     if data.get('timeStamp', False):
         data['https://schema.org/sdDatePublished'] = data.pop('timeStamp')
+
     defaultVocabulary = "https://geojson.org/geojson-ld/geojson-context.jsonld"
-    ldjsonData = {
-        "@context": [defaultVocabulary, *(context or [])],
-        **data
-    }
-    isCollection = identifier is None
-    if isCollection:
-        for i, feature in enumerate(data['features']):
-            featureId = feature.get(
-                'id', None
-            ) or feature.get('properties', {}).get('id', None)
-            if featureId is None:
-                continue
-            # Note: @id or https://schema.org/url or both or something else?
-            if is_url(str(featureId)):
-                feature['id'] = featureId
-            else:
-                feature['id'] = '{}/{}'.format(data['id'], featureId)
+    if not geojsonld:
+        ldjsonData = {
+            "@context": [
+                {
+                "schema":"https://schema.org/",
+                identifier_field: "@id",
+                "type": "@type",
+                },
+                *(context or []),
+                *(geocontext or [])
+            ],
+            **data
+        }
+    else:
+        ldjsonData = {
+            "@context": [
+                defaultVocabulary,
+                *(context or [])
+            ],
+            **data
+        }
+
+    return ldjsonData
```

Comment:
`json.dumps(ldjsonData)`, i.e. this function no longer returns a string - is that correct?

Reply:
I moved that into api.py so the json-ld could refer to `self.pretty_print` (line 1209 of api.py) early on for readability. I can see how this might not fit the specifications of `geojson2geojsonld(` though.